classes/template/PKPTemplateManager.inc.php

```diff
@@ -249,12 +250,78 @@ function setCacheability($cacheability = CACHEABILITY_PUBLIC) {
 	}
 
 	/**
+	 * Compile a LESS stylesheet
+	 *
+	 * @param $name string Unique name for this LESS stylesheet
+	 * @param $lessFile string Path to the LESS file to compile
+	 * @param $args array Optional arguments. SUpports:
+	 *   'baseUrl': Base URL to use when rewriting URLs in the LESS file.
+	 *   'addLess': Array of additional LESS files to parse before compiling
+	 * @return string Compiled CSS styles
+	 */
+	public function compileLess($name, $lessFile, $args = array()) {
+
+		// Load the LESS compiler
+		require_once('lib/pkp/lib/vendor/oyejorge/less.php/lessc.inc.php');
+		$less = new Less_Parser(array( 'relativeUrls' => false ));
+
+		$request = $this->_request;
+
+		// Allow plugins to intervene
+		HookRegistry::call('PageHandler::compileLess', array($request, $less, $name, $lessFile, $args));
```

Comment:
Hook calls are one of the few places where references can be necessary. For the plugin to override any of the values passed in the array, the array contents will need to be specified by reference:

    array(&$request, &$less, &$name, &$lessFile, &$args)